datahub-web-react/src/app/policy/PolicyDetailsModal.tsx

```diff
@@ -98,26 +110,50 @@ export default function PolicyDetailsModal({ policy, visible, onClose, privilege
                         <div>
                             <Typography.Title level={5}>Asset Type</Typography.Title>
                             <ThinDivider />
-                            <PoliciesTag>
-                                {mapResourceTypeToDisplayName(
-                                    policy?.resources?.type || '',
-                                    policiesConfig?.resourcePrivileges || [],
-                                )}
-                            </PoliciesTag>
+                            {(resourceTypes?.length &&
+                                resourceTypes.map((value, key) => {
+                                    return (
+                                        // eslint-disable-next-line react/no-array-index-key
+                                        <PoliciesTag key={`type-${value.value}-${key}`}>
+                                            <Typography.Text>
+                                                {mapResourceTypeToDisplayName(
+                                                    value.value,
+                                                    policiesConfig?.resourcePrivileges || [],
+                                                )}
+                                            </Typography.Text>
+                                        </PoliciesTag>
+                                    );
+                                })) || <PoliciesTag>All</PoliciesTag>}
                         </div>
                         <div>
                             <Typography.Title level={5}>Assets</Typography.Title>
                             <ThinDivider />
-                            {policy?.resources?.resources?.map((urn, key) => {
-                                // TODO: Wrap in a link for entities.
-                                return (
-                                    // eslint-disable-next-line react/no-array-index-key
-                                    <PoliciesTag key={`${urn}-${key}`}>
-                                        <Typography.Text>{urn}</Typography.Text>
-                                    </PoliciesTag>
-                                );
-                            })}
-                            {policy?.resources?.allResources && <PoliciesTag>All</PoliciesTag>}
+                            {(resourceEntities?.length &&
+                                resourceEntities.map((value, key) => {
+                                    return (
+                                        // eslint-disable-next-line react/no-array-index-key
+                                        <PoliciesTag key={`resource-${value.value}-${key}`}>
+                                            <Typography.Text>
+                                                {getDisplayName(value.entity) || value.value}
```

Comment:
minor : 

do value.entity && getDisplayName(value.entity) || value.value (in both)